src/main/java/com/uid2/shared/auth/KeysetSnapshot.java

```diff
@@ -20,5 +21,18 @@ public boolean canClientAccessKey(ClientKey clientKey, KeysetKey key) {
         return keyset.canBeAccessedBySite(clientKey.getSiteId());
     }
 
-    public Map<Integer, Keyset> getAllKeysets() { return keysets; }
+    @Override
+    public boolean canClientAccessKey(ClientKey clientKey, KeysetKey key, MissingAclMode accessMethod) {
+        Keyset keyset = keysets.get(key.getKeysetId());
```

Comment:
1. Why do we need to keep the old "canClientAccessKey(ClientKey clientKey, KeysetKey key)" method?
2. Assuming we do need it, I think this method should have all the functionality, and canClientAccessKey(ClientKey clientKey, KeysetKey key) should simply call canClientAccessKey(ClientKey clientKey, KeysetKey key, MissingAclMode accessMethod) with the appropriate default.
3. Do we have test coverage for both these methods?